spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin/ZipkinSpanListener.java

```diff
@@ -189,17 +182,7 @@ private void addZipkinBinaryAnnotations(io.zipkin.Span.Builder zipkinSpan,
 		}
 	}
 
-	private static long hash(String string) {
-		long h = 1125899906842597L;
-		if (string == null) {
-			return h;
-		}
-		int len = string.length();
-
-		for (int i = 0; i < len; i++) {
-			h = 31 * h + string.charAt(i);
-		}
-		return h;
+	private long valueOrDefault(Long value) {
```

Comment:
One way out is to look at the `event.getSpan().isExportable()` contract. A span without an identifier isn't effectively exportable. we could check ids inside that method. Or, before calling convert, guard on ids being present and don't attempt to report/export spans who have no id. Spans without an id cannot be aggregated into a trace, or correlated with other spans. We have a big TODO around how it is even possible to create id-less spans!